### ISSUE TITLE
[Android][a11y] Widgets that set `Semantics.link` but not `Semantics.linkUrl` are not described as anything

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#172982]
### Steps to reproduce

Internal bug: b/435115601, client `huggsy`. I believe https://github.com/flutter/flutter/pull/162419 is the source of this regression.

1. Run the code sample below on an Android device
2. Turn on TalkBack
3. Navigate to the + button using TalkBack

### Expected results

The widget should be read as "Increment, link, double-tap to activate" or "Increment, button, double-tap to activate".

### Actual results

The widget is read as "Increment, double-tap to activate".

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: Text(
          'Button tapped $_counter time${_counter == 1 ? '' : 's'}.',
          key: Key('CountText'),
        ),
      ),
      floatingActionButton: Semantics(
        container: true,
        link: true,
        label: 'Increment',
        child: Container(
          width: 48,
          height: 48,
          alignment: Alignment.center,
          child: InkWell(
            onTap: _incrementCounter,
            child: const Icon(Icons.add),
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

<img width="1080" height="2092" alt="Image" src="https://github.com/user-attachments/assets/dabe2d3a-d5ed-48cd-b1f5-7110629ed7dc" />

</details>

### Logs

<details open><summary>Logs</summary>

N/A

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

N/A, Google internal client

</details>

**Comment by math1man** [comment]
FYI @chunhtai

**Comment by darshankawar** [comment]
Thanks for the report. Verified with latest sdk versions and on Android and iOS with below results:

1. Android:  `Increment, double-tap to activate`
2. iOS : `increment, link`.

```
stable : 3.32.8.
master : 3.33.0-1.0.pre-1200
```

**Comment by chunhtai** [comment]
The previous behavior will treat it as a button which is not accurate as well, and it requires a url to be treated as link in Android. I can't think of good way to move forward with this issue besides adding more doc/ check to warn user about the use of Semantics(link: true).

As for the customer, I think the correct way will be provide a linkUrl if you are also using link: true

**Comment by math1man** [comment]
We can't add a link URL because the behavior is navigating within the app. We could set `button: true` instead, but that would have undesirable side effects for iOS and Web, which already behave correctly. I think the optimal solution is that the Flutter framework ought to treat this case as a button automatically, the same as it did before https://github.com/flutter/flutter/pull/162419.

**Comment by math1man** [comment]
We're also seeing this on iOS, and on iOS there is no indication that the link is clickable, which is a more serious issue. Internal bug b/445874276. Given the increased severity here, can the priority be increased?

**Comment by chunhtai** [comment]
iOS should treat the semantics.link: true to be a link element. Unless something may have changed in the recent iOS update. will take a look and re-evaluate the priority

**Comment by chunhtai** [comment]
looks like iOS stopped announce `link` when `link: true` is set

**Comment by math1man** [comment]
I tested with the submitted fix and Android still encounters the same issue. I haven't been able to test on iOS.

**Comment by math1man** [comment]
Can you please reopen this issue?

**Comment by darshankawar** [comment]
> I tested with the submitted fix and Android still encounters the same issue. I haven't been able to test on iOS.

I verified on latest master `3.37.0-1.0.pre-395` and see the same results as before, ie:



> * Android:  `Increment, double-tap to activate`
> * iOS : `increment, link`.

**Comment by chunhtai** [comment]
For Android  is WAI, android needs the a linkURL to decorate something as link.

**Comment by math1man** [comment]
I don't think that's technically true. Native Android apps can use [ClickableSpan](http://go/android-dev/reference/android/text/style/ClickableSpan) to configure arbitrary click behaviors on an inline link. But regardless, the behavior for Flutter inline links without URLs used to be that they were treated as a `button` for the purposes of semantics (prior to http://go/android-dev/reference/android/text/style/ClickableSpan), and returning to that behavior seems like a reasonable workaround assuming there is some technical reason why defining them as links isn't possible.

**Comment by chunhtai** [comment]
Thanks for additional info, I will take a closer look

**Comment by chunhtai** [comment]
Clickable Span doesn't work for talkback if used as contentdescription direclty, and there isn't appear to be a role specific for link in talkback, will revert the behavior to be a button for now

**Comment by math1man** [comment]
Thanks for looking into it!

**Comment by math1man** [comment]
Thanks!
